routes/adminapi.js

```diff
@@ -84,6 +87,122 @@ router.post('/saveword', function(req, res) {
+    } else {
+      res.status(400).send('Invalid request format');
+    }
+
+    Promise.all(dbPromises).then((results) => {
+      let response = {Saved: 0, Duplicate: 0, errors: 0};
+      for (var i = 0; i < results.length; i++) {
+        if (results[i] === 'Duplicate' || results[i] === 'Saved') {
+          response[results[i]] += 1;
+        } else response.errors += 1;
+      }
+      res.send(JSON.stringify(response));
+    })
+    .catch((err) => {
+      res.status(500).send('Unknown Error Occurred, some words might have been saved');
+    });
+  });
+});
+
+function saveWord(payload, email) {
+  return new Promise((resolve, reject) => {
+    console.log(payload.word);
+    db.checkWordExists(payload.word, (err, message) => {
+      if (err) return reject(err);
+      if (message === true) {
+        return resolve('Duplicate');
+      }
+      payload.author = email;
+      db.insertWordInCollection(payload, (err, message) => {
+        if (err) return reject(err);
+        resolve(message);
+      });
+    });
+  });
+}
+
+// Endpoint to pick the words from ./routes/wordlist.txt and create game configs
+// curl localhost:8080/adminapi/makegames?wordcount=2
+router.get('/makegames', (req, res, next) => {
+  let wordcount = req.query.wordcount;
+  fs.readFile('./routes/wordlist.txt', 'utf8', function (err, data) {
+    if (err) {
+      res.status(500).send('Error Occurred');
+      return console.log(err);
+    }
+    data = data.replace(/(?:\r\n|\r|\n)/g, '').replace(/( )/g, '');
+    evalute(data, wordcount, res);
+  });
+});
+
+var wordsProcessed = 0;
+function evalute(data, wordcount, res) {
+  let words = data.split(',');
+  let dbLookups = [];
+  for (var i = wordsProcessed; i < words.length && dbLookups.length < wordcount; i++) {
+    let word = words[i];
+    // remove \n which may have crept in the last word
+    if (word.slice(-1) === '\n') word = word.slice(0, word.length - 1);
+    if (word.length < 4) continue;
+
+    dbLookups.push(new Promise((resolve, reject) => {
+      db.checkWordExists(word, (err, found) => {
+        if (err) return reject(err);
+        resolve({'found': found, 'word': word});
+      });
+    }));
+  }
+
+  Promise.all(dbLookups).then((results) => {
+    let resData = {words: []};
+    let pexelCalls = [];
+    for (var i = 0; i < results.length; i++) {
+      if (!results[i].found) {
+        resData.words.push({word: results[i].word, images: []});
+        // API calls to pexel
+        pexelCalls.push(pexelReq(results[i].word));
+        wordsProcessed += 1;
+      }
+    }
+    Promise.all(pexelCalls).then((results) => {
+      for (var j = 0; j < results.length; j++) {
+        if (!results[j].body.hasOwnProperty('photos')) continue;
+        results[j].body.photos.slice(0, 4).forEach((item) => {
+          resData.words[j].images.push(item.src.medium);
+        });
+      }
+      res.send(JSON.stringify(resData));
+    });
+  });
+}
+
+function pexelReq(word) {
+  return request
```

Comment:
use fetch instead, its a standard and is being used everywhere in the project...
check /subscribe